### PR TITLE
Set chat active on being re-invited

### PR DESCRIPTION
**Pull request by cammellos**
If the user deletes/leaves a group chat, the chat is set as not active.
This means that if we are re-invited to the chat it won't be shown to
the user.

This commit changes this behavior so that if we are re-invited to the
chat it is set as active again.

### CODE REVIEW

File: protocol/chat.go [hunk 1]
```diff
@@ -150,7 +150,16 @@ func (c *Chat) MembersAsPublicKeys() ([]*ecdsa.PublicKey, error) {
...
 }
 
-func (c *Chat) updateChatFromProtocolGroup(g *v1protocol.Group) {
+func (c *Chat) updateChatFromProtocolGroup(myID string, g *v1protocol.Group) {
+
+	// Check if we were already in the chat
+	var hadJoined bool
+	for _, member := range c.Members {
+		if myID == member.ID {
+			hadJoined = true
```
Comment (jakubgs) [inline protocol/chat.go]: Wouldn't it be cleaner to have a `func (c *Chat) HasMember(memberId string) bool` method?

File: protocol/messenger_test.go [hunk 2]
```diff
@@ -936,6 +936,73 @@ func (s *MessengerSuite) TestChangeNameGroupChat() {
...
+
+	_, err = theirMessenger.ConfirmJoiningGroup(context.Background(), ourChat.ID)
+	s.NoError(err)
+
+	// Wait for join group event
+	err = tt.RetryWithBackOff(func() error {
+		var err error
+		response, err = s.m.RetrieveAll()
+		if err == nil && len(response.Chats) == 0 {
+			err = errors.New("no joining group event received")
```
Comment (jakubgs) [inline protocol/messenger_test.go]: We use this `RetryWithBackOff()` block a lot in these tests, wouldn't it make sense to refactor it out as a helper for waiting with a custom error message as argument?

**Review by jakubgs — APPROVED** [review]

File: protocol/chat.go [hunk 3]
```diff
@@ -150,7 +150,16 @@ func (c *Chat) MembersAsPublicKeys() ([]*ecdsa.PublicKey, error) {
 	return stringSliceToPublicKeys(publicKeys, true)
 }
 
-func (c *Chat) updateChatFromProtocolGroup(g *v1protocol.Group) {
+func (c *Chat) updateChatFromProtocolGroup(myID string, g *v1protocol.Group) {
```
Comment (Samyoul) [inline protocol/chat.go]: The only real issue I've got is the name of the function. It should probably change as it is no longer `updateChatFromProtocolGroup(g *v1protocol.Group)` its now something like `updateChanges(myID string, g *v1protocol.Group)`.

What do you think?

File: protocol/chat.go (hunk 1 above)
Comment (Samyoul) [reply protocol/chat.go]: I thought this, but a `HasMember()` method also introduces an extra iteration over the `c.Members`

Comment (Samyoul) [reply protocol/chat.go]: Ah sorry ignore me. I'm looking at the wrong place in the code 🤦

Comment (Samyoul) [reply protocol/chat.go]: `func (c *Chat) HasMember(memberId string) bool` would be an improvement.

**Pull request merged**